client/src/helpers/storeSearch.ts

```diff
@@ -0,0 +1,340 @@
+import { DEFAULT_NGRAM_LENGTH } from "src/constants";
+import { Store } from "src/interfaces";
+import { Heap } from "src/packages/Heap";
+
+/**
+ * Preprocesses a string s to use as input for calculateEditDistance().
+ *
+ * @param s
+ */
+export const preprocess = (s: string) => {
+  s = s.toLowerCase(); // make lower case
+  s = s.replace(/[().|[\]{}`]/, ""); // remove the list of chars defined above
+  s = s.replace("&", "and");
+  s = s.replace(",", " ");
+  s = s.replace("-", " ");
+  s = s.replace(/[^\w\s]/, ""); // remove any remaining non-word characters
+  s = s.replace(/\s\s+/g, " "); // replace multiple spaces with a single space
+  s = s.replace(
+    /\w\S*/g,
+    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
+  ); // capital at start of each word and lowercase remaining letters
+  return s;
+};
+
+/**
+ * Runs the Levenshtein distance algorithm with modifications.
+ * Returns an array of the inner edit distance, the number of trailing
+ * insertions, and the number of leading insertions.
+ *
+ * @param s1
+ * @param s2
+ */
+const calculateEditDistance = (s1: string, s2: string) => {
+  const m = s1.length;
```

Comment:
Use better name instead of m and n
```suggestion
  const s1Len = s1.length;
```